### PR TITLE
update: add code for checking algebraically independent parameters am…

### PR DESCRIPTION
**Pull request by iliailmer**
…ong unidentifiable ones

### CODE REVIEW

File: IdentifiabilityODE.mpl [hunk 1]
```diff
@@ -251,18 +273,17 @@ IdentifiabilityODE := proc(system_ODEs, params_to_assess, {p := 0.99, count_solu
   if infolevel > 1 then
     printf("%s %a\n", `Bound D_2 for assessing global identifiability: `, D2):
   end if:
-
   # (b, c) ---------
-  sample := SamplePoint(D2, x_vars, y_vars, u_vars, mu, X_eq, Y_eq, Q):
+  sample := SamplePoint(D2, x_vars, [op(y_vars), op(y_faux)], u_vars, mu, X_eq, Y_eq, Q):
```
Comment (pogudingleb) [inline IdentifiabilityODE.mpl]: Maybe add `y_faux` to `y_vars` at the same place where we add the equations to `Y_eq` for consistency?

File: IdentifiabilityODE.mpl [hunk 2]
```diff
@@ -185,6 +188,20 @@ IdentifiabilityODE := proc(system_ODEs, params_to_assess, {p := 0.99, count_solu
             polys_to_process := new_to_process:
           end do:
         else
+          if sub_non_id then 
+            pivots := {}:
+            for row_idx from 1 to nops(eqs_i) do #nops(theta) do
+                row := rrefJacX[row_idx]:
+                pivot_idx := 1:
+                while row[pivot_idx]=0 and add(row)<>0 do
```
Comment (pogudingleb) [inline IdentifiabilityODE.mpl]: I would say that this logical structure may be confusing. I would suggest to check `add(row) <> 0` just once, and, if false, skip the while loop at all.

File: IdentifiabilityODE.mpl (hunk 1 above)
Comment (iliailmer) [reply IdentifiabilityODE.mpl]: The whole approach was modified, I think this is not an issue anymore.

**Comment by iliailmer** [comment]
I will close this pr in favor of a cleaner one.

**Pull request closed without merging**